third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -645,6 +667,15 @@ impl Interpreter {
                 .map_err(|err| err.to_partial())
             },
             NativeResult::LoadModule { module_name } => {
+                if let Loader::V2(_) = resolver.loader() {
```

Comment:
mhm, why do we need a new check?

Reply:
Below we do `check_dependencies_and_charge_gas`. Current implementation uses annoying `allow_module_failure` only for the first set of `ids` because these dependencies indeed may not exist (e.g., non existent entry function, etc.) but for other dependencies down the closure they always exist. So we optionally check if allow LINKER_ERROR or it is an invariant violation

Reply:
But actually yes, in this implementation I do not think we need this here, good point